Src/SwqlStudio/Utils/CommandLineGenerator.cs

```diff
@@ -7,25 +9,45 @@ namespace SwqlStudio.Utils
 {
     internal static class CommandLineGenerator
     {
-        public static string GetQueryForCurlCmd(string query, ConnectionInfo connection)
+        public static string GetQueryForCurlCmd(string query, ConnectionInfo connection, PropertyBag parameters)
         {
             string creds = QuoteForCmd($"{connection.UserName}:{connection.Password}");
             return $"curl.exe -k -u {creds} {GetUrlForQuery(query, connection)}";
         }
 
-        public static string GetQueryForCurlBash(string query, ConnectionInfo connection)
+        public static string GetQueryForCurlBash(string query, ConnectionInfo connection, PropertyBag parameters)
         {
             string creds = $"{connection.UserName}:{connection.Password}";
             var url = GetUrlForQuery(query, connection);
             Func<string, string> q = QuoteForBash;
             return $"curl -k -u {QuoteForBash(creds)} {QuoteForBash(url)}";
         }
 
-        public static string GetQueryForPowerShellGetSwisData(string query, ConnectionInfo connection)
+        public static string GetQueryForPowerShellGetSwisData(string query, ConnectionInfo connection, PropertyBag parameters)
         {
             Func<string, string> q = QuoteForPowerShell;
+            var paramsObjectString = GetParamsObjectInPowershellFormat(parameters);
             return $"Get-SwisData (Connect-Swis -Hostname {connection.Server} -Username {q(connection.UserName)} " +
-                   $"-Password {q(connection.Password)}) -Query {q(CollapseWhitespace(query))}";
+                   $"-Password {q(connection.Password)}) -Query {q(CollapseWhitespace(query))} -Parameters {paramsObjectString}";
+        }
+
+        private static string GetParamsObjectInPowershellFormat(PropertyBag parameters)
+        {
+            var result = new StringBuilder();
+            var parametersCount = parameters.Count;
+            var currentParameterIndex = 0;
+            foreach (var param in parameters)
+            {
+                decimal tempNum = 0;
+                currentParameterIndex++;
+                var isNumber = decimal.TryParse(param.Value.ToString(), out tempNum);
```

Comment:
You are not actually using `isNumber` here. I think it was probably intended to be used to skip wrapping the value in quotes on the next line, but SWIS doesn't really care. It will coerce the value from a string to a number if it needs to.